astropy/coordinates/builtin_frames/ecliptic_transforms.py

```diff
@@ -31,9 +31,8 @@ def _mean_ecliptic_rotation_matrix(equinox):
     # function and also brings the results closer to what other libraries
```

Comment:
I think with the change we also need to adjust the comment, explaining why this function exists at all! Maybe
```
This code just calls ecm06, which uses the precession matrix according to the
IAU 2006 model, but leaves out nutation. This brings the results closer to what
other libraries give (see https://github.com/astropy/astropy/pull/6508).
```